keras/models.py

```diff
@@ -123,10 +123,7 @@ def get_json_type(obj):
                 '(see keras.io/optimizers).')
         else:
             f.attrs['training_config'] = json.dumps({
-                'optimizer_config': {
-                    'class_name': model.optimizer.__class__.__name__,
-                    'config': model.optimizer.get_config()
-                },
+                'optimizer_config': model.optimizer,
```

Comment:
An optimizer instance is not serializable (it's a Python object). The previous 4 lines were serializing it. Revert this